domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/stoptimes/StopTime.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.domain.entity.stoptimes;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Entity representing a row from stop_times.txt.
+ * See http://gtfs.org/reference/static#stop_timestxt
+ * <p>
+ * This class can not be directly instantiated. User must use {@link StopTimeBuilder} to create a {@link StopTime}
+ * object.
+ */
+public class StopTime {
+
+    @NotNull
+    private final String tripId;
+
+    @Nullable
+    private final String arrivalTime;
+
+    @Nullable
+    private final String departureTime;
+
+    @NotNull
+    private final String stopId;
+
+    private final int stopSequence;
+
+    @Nullable
+    private final String stopHeadsign;
+
+    @NotNull
+    private final PickupType pickupType;
+
+    @NotNull
+    private final DropOffType dropOffType;
+
+    @Nullable
+    private final Float shapeDistTraveled;
+
+    @NotNull
+    private final Timepoint timePoint;
+
+    private StopTime(@NotNull String tripId,
+                     @Nullable String arrivalTime,
+                     @Nullable String departureTime,
+                     @NotNull String stopId,
+                     int stopSequence,
+                     @Nullable String stopHeadsign,
+                     @NotNull PickupType pickupType,
+                     @NotNull DropOffType dropOffType,
+                     @Nullable Float shapeDistTraveled,
+                     @NotNull Timepoint timepoint) {
+        this.tripId = tripId;
+        this.arrivalTime = arrivalTime;
+        this.departureTime = departureTime;
+        this.stopId = stopId;
+        this.stopSequence = stopSequence;
+        this.stopHeadsign = stopHeadsign;
+        this.pickupType = pickupType;
+        this.dropOffType = dropOffType;
+        this.shapeDistTraveled = shapeDistTraveled;
+        this.timePoint = timepoint;
+    }
+
+    @NotNull
+    public String getTripId() {
+        return tripId;
+    }
+
+    @Nullable
+    public String getArrivalTime() {
+        return arrivalTime;
+    }
+
+    @Nullable
+    public String getDepartureTime() {
+        return departureTime;
+    }
+
+    @NotNull
+    public String getStopId() {
+        return stopId;
+    }
+
+    public int getStopSequence() {
```

Comment:
GTFS stop_times.txt isn't guaranteed to be ordered, so at some point we'll need to sort `StopTime` objects - grouped by `tripId`, ordered by increasing `stopSequence`. To support this you may want to implement `Comparable` - see https://www.geeksforgeeks.org/comparable-vs-comparator-in-java/ for an example. We can revisit this later unless you have a good idea how this entity will be used within business validation rules now.

Reply:
Liked to https://github.com/MobilityData/gtfs-validator/pull/70#discussion_r401030021

Reply:
See 472fdf85ede58620d42f1af7d39684e04668d175